src/test/unit_tests/test_nc_iam_basic_integration.js

```diff
@@ -0,0 +1,234 @@
+'use strict';
+
+const path = require('path');
+const _ = require('lodash');
+const mocha = require('mocha');
+const assert = require('assert');
+const fs_utils = require('../../util/fs_utils');
+const { TMP_PATH, generate_nsfs_account, get_new_buckets_path_by_test_env, generate_iam_client,
+         get_coretest_path } = require('../system_tests/test_utils');
+const { ListUsersCommand, CreateUserCommand, GetUserCommand, UpdateUserCommand, DeleteUserCommand,
+        ListAccessKeysCommand, CreateAccessKeyCommand, GetAccessKeyLastUsedCommand,
+        UpdateAccessKeyCommand, DeleteAccessKeyCommand } = require('@aws-sdk/client-iam');
+const { ACCESS_KEY_STATUS_ENUM } = require('../../endpoint/iam/iam_constants');
+
+const coretest_path = get_coretest_path();
+const coretest = require(coretest_path);
+const setup_options = { should_run_iam: true, https_port_iam: 7005, debug: 5 };
+coretest.setup(setup_options);
+const { rpc_client, EMAIL, get_current_setup_options, stop_nsfs_process, start_nsfs_process } = coretest;
+
+const CORETEST_ENDPOINT_IAM = coretest.get_iam_https_address();
+
+const config_root = path.join(TMP_PATH, 'test_nc_iam');
+// on NC - new_buckets_path is full absolute path
+// on Containerized - new_buckets_path is the directory
+const new_bucket_path_param = get_new_buckets_path_by_test_env(config_root, '/');
+
+let iam_account;
+
+mocha.describe('IAM basic integration tests - happy path', async function() {
+    this.timeout(50000); // eslint-disable-line no-invalid-this
+
+    mocha.before(async () => {
+        // we want to make sure that we run this test with a couple of forks (by default setup it is 0)
+        const current_setup_options = get_current_setup_options();
+        const same_setup = _.isEqual(current_setup_options, setup_options);
+        if (!same_setup) {
+            console.log('current_setup_options', current_setup_options, 'same_setup', same_setup);
+            await stop_nsfs_process();
+            await start_nsfs_process(setup_options);
+        }
+
+        // needed details for creating the account (and then the client)
+        await fs_utils.create_fresh_path(new_bucket_path_param);
+        await fs_utils.file_must_exist(new_bucket_path_param);
+        const res = await generate_nsfs_account(rpc_client, EMAIL, new_bucket_path_param, { admin: true });
+        iam_account = generate_iam_client(res.access_key, res.secret_key, CORETEST_ENDPOINT_IAM);
+    });
+
+    mocha.after(async () => {
+        fs_utils.folder_delete(`${config_root}`);
+    });
+
+    mocha.describe('IAM User API', async function() {
+        const username = 'Asahi';
+        const new_username = 'Botan';
+
+        mocha.it('list users - should be empty', async function() {
+            const input = {};
+            const command = new ListUsersCommand(input);
+            const response = await iam_account.send(command);
+            assert.equal(response.$metadata.httpStatusCode, 200);
```

Comment:
expect the status code, can you check the actual response?

Reply:
I don't see a point in a test that is sending list accounts and just checking that the server responded 200 without checking what's the actual response

Reply:
We have a basic check `assert.equal(response.Users.length, 0);` so there is always as I understand a basic check on the response.